modules/extensions-mule-support/src/main/java/org/mule/runtime/module/extension/mule/api/processor/MuleSdkRaiseErrorProcessor.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.mule.api.processor;
+
+import static org.mule.runtime.api.component.ComponentIdentifier.builder;
+
+import org.mule.runtime.api.component.ComponentIdentifier;
+import org.mule.runtime.core.api.processor.AbstractRaiseErrorProcessor;
+
+/**
+ * Processor capable of raising errors within a Mule Operation's body on demand, given a type and optionally a message.
+ *
+ * @since 4.5
+ */
+public final class MuleSdkRaiseErrorProcessor extends AbstractRaiseErrorProcessor {
+
+  @Override
+  protected ComponentIdentifier calculateErrorIdentifier(String typeId) {
+    return builder().namespace("THIS").name(typeId).build();
```

Comment:
Have we checked what would happen if someone tries to do something like this?
`<operation:raise-error type="THIS:CUSTOM" />`
Or even:
`<operation:raise-error type="OTHER:CUSTOM" />`

Shouldn't we have a test for those cases showing the expected outcome (maybe error in both, or maybe ignoring the namespace in the first one)?

Reply:
Also, we should probably extract the string "THIS" to a shared constant so can track its usages more easily.
Maybe making this one public or moving it to a constants class: https://github.com/mulesoft/mule/blob/eb7c7bb88896929865674e4f36aed563311ff565/modules/extensions-mule-support/src/main/java/org/mule/runtime/module/extension/mule/internal/loader/parser/MuleSdkExtensionModelParser.java#L55